Sources/Xcodeproj/generate().swift

```diff
@@ -134,3 +135,27 @@ func open(_ path: AbsolutePath, body: ((String) -> Void) throws -> Void) throws
     // Write the real file.
     try localFS.writeFileContents(path.asString, bytes: stream.bytes)
 }
+
+/// Finds directories that will be added as blue folder
+/// Excludes hidden directories and Xcode projects and directories that contains source code
+func findDirectoryReferences(path: String) throws -> [AbsolutePath] {
```

Comment:
One improvement would be to make `path` be an `AbsolutePath` here (it looks from the call site as if it's always intended to be an absolute path here).